### PR TITLE
1048 local storage to keep track of visible layers

### LINKED ISSUE #1048: Use browser's local storage to keep track of which layers are visible
* Save node and link layers to local storage

PR: https://github.com/HSLdevcom/jore-map-ui/pull/1071

### PR DESCRIPTION
**Pull request by Divinare**
Closes #1048 

* Node / link layers stored in local storage
* Added a temporary lazy load functionality to prevent lag. This issue should fix this hack: https://github.com/HSLdevcom/jore-map-ui/issues/1070

### CODE REVIEW

File: src/stores/networkStore.ts [hunk 1]
```diff
@@ -128,16 +136,65 @@ export class NetworkStore {
...
+    @action
+    public saveMapLayers() {
         this._savedMapLayers = this._visibleMapLayers;
     }
+
+    // TODO: Remove this lazy load hack when map's initial position is immediately at target element after page reload
+    private lazyLoadLocalStorageVisibleLayers = async () => {
+        setTimeout(() => {
+            const localStorageVisibleLayers = LocalStorageHelper.getItem('visible_layers');
+            if (!Array.isArray(localStorageVisibleLayers)) return [];
```
Comment (ahjyrkia) [inline src/stores/networkStore.ts]:
```
const layers: MapLayer[] = [];

if (!Array.isArray(localStorageVisibleLayers)) return layers;
```

File: src/stores/networkStore.ts [hunk 2]
```diff
@@ -128,16 +136,65 @@ export class NetworkStore {
...
+    if (mapLayer !== MapLayer.node && mapLayer !== MapLayer.link) {
+        throw `Unsupported mapLayer: ${mapLayer}`;
+    }
+    const mapLayerName = mapLayer === MapLayer.node ? 'node' : 'link';
+    const localStorageVisibleLayers = LocalStorageHelper.getItem('visible_layers');
+    let layers: string[] = Array.isArray(localStorageVisibleLayers)
+        ? localStorageVisibleLayers
+        : [];
+    if (isVisible) {
+        if (layers.indexOf(mapLayerName) === -1) {
```
Comment (ahjyrkia) [inline src/stores/networkStore.ts]: ```if (layers.includes(mapLayerName)) layers.push(mapLayerName);```

Includes is a bit cooler than indexOf?

**Comment by ahjyrkia** [comment]
Reviewed. Few minor changes requested

**Comment by Divinare** [comment]
Fixes done

**Pull request merged**